advanced_alchemy/repository/_sync.py

```diff
@@ -94,6 +97,9 @@ def __init__(
             msg = "Session improperly configure"
             raise ValueError(msg)
         self._dialect = self.session.bind.dialect
+        self._prefer_any_clause = any(
+            self._dialect.name == engine_type for engine_type in self.prefer_any_clause_engines or ()
+        )
```

Comment:
What if `prefer_any_clause_engines` was a set?

```suggestion
        self._prefer_any_clause = self._dialect.name in self.prefer_any_clause_engines
```

Reply:
I had it as a set initially, but I wa getting errors about it being unhashable.  That was the reason for it to go to a `tuple`.  A `set` actually made more sense to me.

Reply:
I see we retrieve the dialect from the session:

https://github.com/jolt-org/advanced-alchemy/blob/48265257ccd810960f51b4ad5bd83e14d4abd469/advanced_alchemy/repository/_async.py#L95C45-L95C45

`Session.bind` is typed `Optional[Union[Engine, Connection]]` [ref](https://github.com/sqlalchemy/sqlalchemy/blob/efe4025c068d6746637db362245ab88ce7dfe785/lib/sqlalchemy/orm/session.py#L1429C46-L1429C46).

`Engine.dialect` is just `Dialect` [ref](https://github.com/sqlalchemy/sqlalchemy/blob/efe4025c068d6746637db362245ab88ce7dfe785/lib/sqlalchemy/engine/base.py#L2935), and `Connection.dialect` should be same [ref](https://github.com/sqlalchemy/sqlalchemy/blob/efe4025c068d6746637db362245ab88ce7dfe785/lib/sqlalchemy/engine/base.py#L141C18-L141C18).

`Dialect.name` is typed `str` [ref](https://github.com/sqlalchemy/sqlalchemy/blob/efe4025c068d6746637db362245ab88ce7dfe785/lib/sqlalchemy/engine/interfaces.py#L664). So hard to see where the unhashable error might be coming from. You could try `reveal_type(self._diaclect.name)` and `reveal_type(self.prefer_any_clause_engines)` in the scope that the error was originating from to see if that turns up anything.